fcrepo-persistence-ocfl/src/test/java/org/fcrepo/persistence/ocfl/OCFLPersistentStorageSessionTest.java

```diff
@@ -195,4 +223,243 @@ public void getTriplesOfBinaryDescription() throws Exception {
+        when(objectSession1.commit(eq(CommitOption.UNVERSIONED))).thenReturn(Instant.now().toString());
+        //mock failure on commit for the second object session
+        when(mockSessionFactory.create(eq(ocfObjectId2), anyString())).thenReturn(objectSession2);
+        when(objectSession2.commit(eq(CommitOption.UNVERSIONED))).thenThrow(PersistentStorageException.class);
+
+        final PersistentStorageSession session1 = createSession(index, mockSessionFactory);
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        //perform the create rdf operation
+        session1.persist(rdfSourceOperation2);
+
+        //get triples should now fail because the session is effectively closed.
+        try {
+            session1.commit(CommitOption.UNVERSIONED);
+            fail("session1.commit(...) invocation should fail.");
+        } catch (PersistentStorageException ex) {
+            //attempted rollback should also fail:
+            try {
+                session1.rollback();
+                fail("session1.rollback(...) invocation should fail.");
+            } catch (final PersistentStorageException e) {
+                assertTrue("exception does not contain expected text",
+                        e.getMessage().contains("already committed to the unversioned "));
+            }
+        }
+    }
+
+    @Test
+    public void getTriplesFailsIfCommitHasAlreadyStarted() throws PersistentStorageException {
+        when(mapping.getOcflObjectId()).thenReturn(OCFL_OBJECT_ID);
+        when(mapping.getParentFedoraResourceId()).thenReturn(parentId);
+        when(index.getMapping(resourceId)).thenReturn(mapping);
+
+        final Node resourceUri = createURI(resourceId);
+
+        when(rdfSourceOperation.getTriples()).thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getServerManagedProperties())
+                .thenReturn(new DefaultRdfStream(resourceUri, Stream.empty()));
+        when(rdfSourceOperation.getResourceId()).thenReturn(resourceId);
+        when(rdfSourceOperation.getType()).thenReturn(CREATE);
+
+        //mock success on commit for the first object session
+        when(mockSessionFactory.create(eq(OCFL_OBJECT_ID), anyString())).thenReturn(objectSession1);
+
+        //pause on commit for 1 second
+        when(objectSession1.commit(any(CommitOption.class))).thenAnswer(new Answer<String>() {
+            @Override
+            public String answer(final InvocationOnMock invocationOnMock) throws Throwable {
+                Thread.sleep(1000);
+                return null;
+            }
+        });
+
+        final PersistentStorageSession session1 = createSession(index, mockSessionFactory);
+        //persist the two operations
+        session1.persist(rdfSourceOperation);
+
+        new Thread(() -> {
+            try {
+                session1.commit(CommitOption.UNVERSIONED);
+            } catch (PersistentStorageException e) {
```

Comment:
If this commit fails that would be a problem wouldn't it? Maybe call `fail()` in the catch block or throw a runtime exception

Reply:
Okay.